### PR TITLE
fix(deploy): adapt forge script to deploy diamond

### LINKED ISSUE #605: Diamond refactoring  / merge deployment script
After merging #600 we can have only one deployment script as the previous tokens has been deleted 
- [ ]  merge the  two deployment script to have only one that act as the actual forge script for development but that will deploy

### PR DESCRIPTION
**Pull request by zgorizzo69**
Resolves #605 

you can test with 
```
yarn start:anvil
```
in another term
```
yarn deploy:development
```

### CODE REVIEW

File: packages/contracts/scripts/deploy/dollar/solidityScripting/02_UbiquityDollarToken.s.sol [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
+        IManager.deployStableSwapPool(
             curveFactory,
             basePool,
             curve3PoolToken,
             10,
             5000000
         );
 
-        metapool = manager.stableSwapMetaPoolAddress();
+        metapool = IManager.stableSwapMetaPoolAddress();
```
Comment (0x4007) [inline packages/contracts/scripts/deploy/dollar/solidityScripting/02_UbiquityDollarToken.s.sol:37]: Why do you prefix with `I` now?

File: packages/contracts/scripts/deploy/dollar/solidityScripting/05_StakingShare.s.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+contract StakingShareScript is CreditScript {
+    function run() public virtual override {
+        super.run();
+        vm.startBroadcast(deployerPrivateKey);
+
+        // grant diamond token minter rights
+        IAccessCtrl.grantRole(STAKING_SHARE_MINTER_ROLE, address(diamond));
+        // add staking shares
+        string
+            memory uri = "https://bafybeifibz4fhk4yag5reupmgh5cdbm2oladke4zfd7ldyw7avgipocpmy.ipfs.infura-ipfs.io/";
```
Comment (0x4007) [inline packages/contracts/scripts/deploy/dollar/solidityScripting/05_StakingShare.s.sol:15]: I just remembered this doesn't work. We should consider hosting off-chain and on our own website!

Comment (zgorizzo69) [reply packages/contracts/scripts/deploy/dollar/solidityScripting/05_StakingShare.s.sol:15]: Sure what URI should we use ?

**Comment by 0x4007** [comment]
What `.env` properties exactly did you test with? I would love to test both what you used exactly, and then test with other property values.

File: packages/contracts/scripts/deploy/dollar/solidityScripting/02_UbiquityDollarToken.s.sol (hunk 1 above)
Comment (zgorizzo69) [reply packages/contracts/scripts/deploy/dollar/solidityScripting/02_UbiquityDollarToken.s.sol:37]: Like in the test IxxxFacet is the facet implementation through the diamond proxy whereas xxxFacet is the implementation of the facet (so not through the proxy)

**Comment by zgorizzo69** [comment]
> What `.env` properties exactly did you test with? I would love to test both what you used exactly, and then test with other property values.

Did you try copy pasting and renaming the .env.sample ? Just so I know if there is some env missing but AFAIK we need an rpc and a privatekey

**Comment by 0x4007** [comment]
The RPC should default to our custom in house one and the private key can I generate any random key? e.g. 
```
PUBLIC_KEY = 0x729340388d5a26c9444fd49a0d3473d2da3677b8
PRIVATE_KEY = fb01b9e8f3dbfbfb3b76405f18ecfbaea780d2d557592ed108e005beebdfb07d
```

**Comment by zgorizzo69** [comment]
> The RPC should default to our custom in house one and the private key can I generate any random key? e.g.
> 
> ```
> PUBLIC_KEY = 0x729340388d5a26c9444fd49a0d3473d2da3677b8
> PRIVATE_KEY = fb01b9e8f3dbfbfb3b76405f18ecfbaea780d2d557592ed108e005beebdfb07d
> ```
The private key used must have some ether on the fork chain by anvil

**Comment by 0x4007** [comment]
@rndquu is it possible if you help me test/review? I moved to a new country and have been very busy trying to find a place to live, so my focus time is extremely limited these days. I don't mean to be a blocker so I'm hoping to delegate this review.

**Comment by rndquu** [comment]
> @rndquu is it possible if you help me test/review? I moved to a new country and have been very busy trying to find a place to live, so my focus time is extremely limited these days. I don't mean to be a blocker so I'm hoping to delegate this review.

I will check the PR today

File: packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+        selectorsOfCollectableDustFacet.push(
+            collectableDustFacet.addProtocolToken.selector
+        );
+        selectorsOfCollectableDustFacet.push(
+            collectableDustFacet.removeProtocolToken.selector
+        );
+        selectorsOfCollectableDustFacet.push(
+            collectableDustFacet.sendDust.selector
+        );
+        // Chef
```
Comment (rndquu) [inline packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol:400]: Missing selector for https://github.com/ubiquity/ubiquity-dollar/blob/fix/deploy-script/packages/contracts/src/dollar/facets/ChefFacet.sol#L37

**Review by rndquu — CHANGES_REQUESTED** [review]
Looks fine overall, migration scripts are working fine

There is 1 missing selector so @zgorizzo69 pls add it

We could also create a task for automatic selector populating [here](https://github.com/ubiquity/ubiquity-dollar/blob/fix/deploy-script/packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol#L268) because it is easy to miss one. Foundry can read from json files so perhaps we could read the build ABIs and automatically populate selectors for all facets.

File: packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol (hunk 3 above)
Comment (zgorizzo69) [reply packages/contracts/scripts/deploy/dollar/solidityScripting/01_Diamond.s.sol:400]: good catch I added it !

**Review by rndquu — APPROVED** [review]

**Comment by rndquu** [comment]
@zgorizzo69 @pavlovcik notice the issue regarding automatic selector population https://github.com/ubiquity/ubiquity-dollar/issues/616

**Comment by 0x4007** [comment]
Removed myself delegated to @rndquu to make the merge decision

File: packages/contracts/scripts/deploy/dollar/solidityScripting/05_StakingShare.s.sol (hunk 2 above)
Comment (rndquu) [reply packages/contracts/scripts/deploy/dollar/solidityScripting/05_StakingShare.s.sol:15]: I'll fix it

**Pull request merged**